src/components/SetUp/FirstForm.jsx

```diff
@@ -17,18 +17,47 @@ import {
 import { useNavigate } from 'react-router-dom';
 import LOGO from './fph_logo.png';
 import { useAuth } from '../../contexts/AuthContext';
+import { useBackend } from '../../contexts/BackendContext';
 import PropTypes from 'prop-types'; // Import PropTypes
 
 const FirstForm = ({ admin, nextStep }) => {
-  const { isAdmin } = useAuth();
+  const { isAdmin, currentUser } = useAuth();
+  const { backend } = useBackend();
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [confirmPassword, setConfirmPassword] = useState('');
-  const { signup } = useAuth();
+  const [id, setId] = useState('');
+  const { signup, logout } = useAuth();
   const [loading, setLoading] = useState(false);
   const navigate = useNavigate();
   const toast = useToast();
 
+  useEffect(() => {
+    const setUp = async () => {
+      const urlParams = new URLSearchParams(window.location.search);
+      const idFromUrl = urlParams.get('id');
+      setId(idFromUrl);
+      await logout();
+    };
+
+    setUp();
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'logout'. Either include it or remove the dependency array.